djed-sdk/src/web3.js

```diff
@@ -1,12 +1,35 @@
 import Web3 from "web3";
 import djedArtifact from "./artifacts/DjedABI.json";
 import coinArtifact from "./artifacts/CoinABI.json";
+import oracleArtifact from "./artifacts/OracleABI.json";
 import { convertInt,web3Promise } from "./helpers";
 
-export const DjedInstance = async (BLOCKCHAIN_URI, DJED_ADDRESS) => {
+ const getWeb3 = (BLOCKCHAIN_URI) =>
+  new Promise(async (resolve, reject) => {
+    if (window.ethereum) {
+      try {
+        const web3 = new Web3(BLOCKCHAIN_URI);
+        resolve(web3);
+      } catch (error) {
+        reject(error);
+      }
+    } else {
+      reject("Install Metamask");
+    }
+  });
+
+  const getOracleContract = (web3, oracleAddress, msgSender) => {
+    const oracle = new web3.eth.Contract(oracleArtifact.abi, oracleAddress, {
+      from: msgSender
+    });
+    return oracle;
+  };
+
+export const DjedInstance = async (BLOCKCHAIN_URI, DJED_ADDRESS,msgSender) => {
```

Comment:
Can we move this to `Djed.js`?

Reply:
@Tanya-ruby , I think this wouldn't be a good idea, because this file contains functions to interact with the `web3` library. What we need is to remove Djed-specific functionality from this file and move it to more suitable files.

Reply:
My suggestion is to rename files and folders as follows:

* the "coins" folder should be renamed "djed", because many things in that folder are related to Djed and not only to the coins.

*the "system.js" and "constants.js" file should be moved to the above-mentioned "djed" folder.

* functions related to the oracle should be in a file outside the "djed" folder.

Reply:
# Project Structure

```bash
djed-sdk/
├── djed/
│   ├── djedInstance.js (or djedUtils)# this will fetch the djed contracts,decimals,coin contracts
│   ├── reserveCoin.js
│   ├── stableCoin.js
│   ├── tradeUtils.js
│   └── system.js  # system.js is now under the 'djed' folder
├── helpers.js
├── index.js
├── oracle/
│   ├── index.js
│   └── oracle.js # fetch the oracle address and contract
└── web3.js # setup the web3 instance
```
 @ceilican I've changed the structure to this,so we can also move the constants.js file under the djed,right?
 What do you think is more appropriate naming for the djedInstance.js?